### PR TITLE
Pull city and event info from API into combined city page

### PR DESCRIPTION
**Pull request by parth-vora**
Tried linking data from API instead of hardcoded into Javascript. Questions about the following:

1) Whether the events will pull correctly since it is also coming from a separate location.
2) If the same check error about the async function is relevant for the city site. 
3) If the data is pulling from the API correctly as opposed to the hardcoded data

### CODE REVIEW

File: src/main/webapp/city.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  var cityId = null;
+
+  const citiesResponse = await fetch("/api/city.html?id=" + id);
+  const cities = await citiesResponse.json();
+
+  for (let city in cities) {
+    if ((cities[city]["id"] = id)) {
+      cityId = cities[city]["id"];
+    }
+  }
```
Comment (josueetcom) [inline src/main/webapp/city.js]: Per the [API docs](https://stoplight.io/p/docs/gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1%7BcityID%7D/get?srn=gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1{cityID}/get&group=api-schema) the actual path to get the object for a specific city is `/api/cities/{cityID}` so this is as simple as:
```suggestion
  const cityResponse = await fetch("/api/cities/" + id);
  const city = await cityResponse.json();
```

File: src/main/webapp/city.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  displayCityResults(
+    cities[city],
+    eventName,
+    eventDescription,
+    eventDate,
+    eventLocation,
+    eventPricing
+  );
```
Comment (josueetcom) [inline src/main/webapp/city.js]: Per the [API](https://stoplight.io/p/docs/gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1%7BcityID%7D/get?srn=gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1{cityID}/get&group=api-schema) `city.events` should already contain the events for that city. So I would just do:
```suggestion
  displayCityResults(city);
```

File: src/main/webapp/city.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  });
+
+  /**
+   * Loads seach paramerter.
+   */
+  function loading() {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const query = urlParams.get("q");
+    searching(query);
```
Comment (josueetcom) [inline src/main/webapp/city.js]: `id` would be a better name for the city id parameter

File: src/main/webapp/city.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            eventLocation[i] +
+            ", Pricing: " +
+            eventPricing
+        )
+      );
+      eventsElement.appendChild(item);
+    }
+    document.getElementById("list-of-events").appendChild(eventsElement);
+  }
+}
```
Comment (josueetcom) [inline src/main/webapp/city.js]: Per the [API](https://stoplight.io/p/docs/gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1%7BcityID%7D/get?srn=gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1{cityID}/get&group=api-schema) `city.events` should be a list of event objects so you can do:
```suggestion
  function displayCityResults(city) {
    const cityElement = document.createElement("H1");
    cityElement.innerText = city.name;
    document.getElementById("city-name").appendChild(cityElement);

    const cityDescElement = document.createElement("P");
    cityDescElement.innerText = city.description;
    document.getElementById("city-description").appendChild(cityDescElement);

    var eventsElement = document.createElement("ul");
    for (var event in city.events) {
      var item = document.createElement("li");
      item.appendChild(
        document.createTextNode(
          "Name: " +
            event.name +
            ", Description: " +
            event.description +
            ", Date: " +
            event.date +
            ", Location: " +
            event.location +
            ", Pricing: " +
            event.pricing
        )
      );
      eventsElement.appendChild(item);
    }
    document.getElementById("list-of-events").appendChild(eventsElement);
  }
}
```

File: src/main/webapp/city.html [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        We can't wait to see you travel to your destination. Here is a brief description to get you excited for your
+        trip!
+      </p>
+      <br />
+      <p id="city-description"></p>
+      <br />
+      <br />
+      <br />
+      <p>While you are there, check out these events!</p>
+      <br />
```
Comment (josueetcom) [inline src/main/webapp/city.html]: What happened to the dos/donts?

**Review by josueetcom — CHANGES_REQUESTED** [review]

File: src/main/webapp/city.js (hunk 1 above)
Comment (parth-vora) [reply src/main/webapp/city.js]: thanks this makes a lot more sense now.

File: src/main/webapp/city.html (hunk 5 above)
Comment (parth-vora) [reply src/main/webapp/city.html]: they were moved to the country html, I will make sure to include them in that one.

File: src/main/webapp/city.js [hunk 6]
```diff
@@ -2,95 +2,52 @@
...
-            eventPricing
-        )
-      );
-      eventsElement.appendChild(item);
-    }
-    document.getElementById("list-of-events").appendChild(eventsElement);
+/**
+ * @param {results} results
+ */
+function displayCityResults(city) {
```
Comment (josueetcom) [inline src/main/webapp/city.js]: You can add a `fetch` call to `"/api/countries?id=" + city.countryId` to get information about the country (e.g. dos/donts)

**Review by josueetcom — DISMISSED** [review]
Feel free to submit this as is!

I imagine the country page javascript will look almost the same, except invoking the countries API instead (along with cities API if you'd like to call `GET /api/cities` with the [`countryId`](https://github.com/gferioli0418/travelbud/blob/master/src/main/java/com/google/sps/travelbud/CityServlet.java#L41) parameter)

**Review by gferioli0418 — COMMENTED** [review]
LGTM

File: src/main/webapp/city.js [hunk 7]
```diff
@@ -3,9 +3,11 @@
  */
 async function searching(id) {
   const cityResponse = await fetch("/api/cities/" + id);
+  const countryResponse = await fetch("/api/countries/" + city.countryId);
   const city = await cityResponse.json();
```
Comment (josueetcom) [inline src/main/webapp/city.js]: Gotta switch these two lines
```suggestion
  const city = await cityResponse.json();
  const countryResponse = await fetch("/api/countries/" + city.countryId);
```

File: src/main/webapp/city.js [hunk 8]
```diff
@@ -25,15 +27,41 @@ function loading() {
...
+    var item = document.createElement("li")
+    item.appendChild(dontTip)
+    dontElement.appendChild(item);
+  }
+  document.getElementById("donts").appendChild(dontElement);
+
+  // add language spoken to "dos" list 
+  var languageElement = document.createElement("li")
+  languageElement.appendChild("Learn a few important words in " + country.languages)
+  document.getElementById("language").appendChild(languageElement);
```
Comment (josueetcom) [inline src/main/webapp/city.js]: Nice!

**Review by josueetcom — APPROVED** [review]
Just gotta switch those two lines otherwise LGTM!

**Comment by josueetcom** [comment]
> 1. Whether the events will pull correctly since it is also coming from a separate location.
Yes the GetCity API should return the events together with the city JSON object.
> 2. If the same check error about the async function is relevant for the city site.
Yes. I'm not sure quite why it complains about that. Side note, it looks like there are some JS style issues though. I think `make pretty` fixes some of those.
> 3. If the data is pulling from the API correctly as opposed to the hardcoded data
Hopefully this was answered in our team chat, but pulling from the API is a bit tricky if there isn't any data. Take a look at one of the last messages I posted for instructions on how to initialize your data

**Review by josueetcom — CHANGES_REQUESTED** [review]
Gonna mark changes as requested so the order of the two lines is switched

**Comment by josueetcom** [comment]
FYI, following up on the `async function` issue, looks like it was an eslint error due to the version of JavaScript it was linting against. #26 should fix this so don't worry about this particular error (you can fix it by copying the changes from #26 though)

**Review by josueetcom — DISMISSED** [review]

**Review by josueetcom — APPROVED** [review]

**Comment by josueetcom** [comment]
Great job!

**Pull request merged**